### DISCUSSION TITLE
Feature Request: Suggestion async items call in progress status

### DISCUSSION THREAD

**Discussion by greg-kinasa** [ueberdosis/tiptap#5163] [Feature Requests]
### Description

It would be great if there was a re-render and status prop that identified whether the async call to `items` is in progress.

### Use Case

A user types a shortcode that triggers a server request in order to retrieve the list of suggestions to be displayed.  This lookup  takes non-zero time and showing in an in progress view would provide a better experience.  

This could be solved with either a status variable passed as a prop or callbacks that the component consumer could use in order to manage the state themselves.

### Type

New feature

**Comment by greg-kinasa** [comment]
I ended up solving this with a custom node view and a `loading` attribute on the node.  I update the flag like this:

```
editor.commands.updateAttributes(QUERY_ACTION_NODE_NAME, {
       loading: true,
});
```

Then my React node view shows or hides a progress spinner.

**Comment by tomas-hruby** [comment]
Hi, i think i have the same problem as you and i don't understand your solution.
I want to render the spinner when i write `@`, but the onStart funtion is called only when the data arrive to `items`. How is your solution dealing with this problem please?

**Reply by greg-kinasa** [reply]
Our UX flow is slightly different in that we begin with a static suggestion list before moving into a more dynamic experience.  We've recently moved away from `Suggestion` for our dynamic UX because trying to overload the UI through the `Suggestion` APIs was becoming too convoluted.  

That said, take a look at the callbacks supported.  `onBeforeStart` seems like it could be a good candidate – it's called before the call to `items`.

https://github.com/ueberdosis/tiptap/blob/716c5ef53b7cbcef31c3bf8bd3981a8eef888692/packages/suggestion/src/suggestion.ts#L241

If you're trying to trigger the spinner every time the query changes - as in, the user types, you query and display results – then you might just put the loading flag changes in your `items` call.

**Reply by tomas-hruby** [reply]
This `onBeforeStart` was exactly what i needed. I am surprised it is not mentioned in the documentation at all. Thank you very much.
